curriculum/challenges/english/25-front-end-development/workshop-library-manager/67116d7584d0b469b14579c1.md

```diff
@@ -0,0 +1,122 @@
+---
+id: 67116d7584d0b469b14579c1
+title: Step 4
+challengeType: 1
+dashedName: step-4
+---
+
+# --description--
+
+The rest of the objects representing the books have been filled in for you. You can take a look at them.
+
+Now, you should start working on getting several common parts of the books.
+
+Start by displaying all the books. Create a `displayBooks` function with a `catalog` parameter. The function should return a string that contains the `title`, `author`, and `pages` of all the books in the `library` array.
+
+# --hints--
+
+You should create a `displayBooks` function.
+
+```js
+assert.isFunction(displayBooks)
+```
+
+Your `displayBooks` function should have a `catalog` parameter.
+
+```js
+assert.match(displayBooks.toString(), /catalog/)
+```
+
+Your `displayBooks` function should use an array method.
```

Comment:
I would be more specific here because we aren't looking for any array method but  we looking for `map`, `reduce`, etc.

```suggestion
Your `displayBooks` function should use a higher order function. Ex. (`filter`, `map`, `reduce`)
```